AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -409,7 +407,33 @@ extension CardViewController: CardViewControllerProtocol {
 
 extension CardViewController {
     private func openCardScanner() {
-        // TODO: Use CardScanning module
+        let scannerNavigationController = UINavigationController()
+        if let scannerViewController = AdyenCardScanner.CardScanner.createCardScanner(completion: { [weak self] result in
+            switch result {
+            case let .success(card):
+                scannerNavigationController.dismiss(animated: true)
+                self?.processScannedCard(card)
+            case let .failure(error):
+                // TODO: Error handling
+                print("Scanner error: \(error)")
+            }
+        }) {
+            scannerNavigationController.setViewControllers(
+                [scannerViewController],
+                animated: false
+            )
+            presentViewController(scannerNavigationController, animated: true)
+        }
+    }
+    
+    private func processScannedCard(_ card: AdyenCardScanner.CreditCard) {
+        items.numberContainerItem.numberItem.value = card.number ?? ""
```

Comment:
Let's add the same function logic as setting the expiry date? Even better to add it to `FormCardNumberContainerItem` as a convenience and pass the value down to its CardNumberItem.

Reply:
Yes we can go for it. The way we expose the whole chain of logic elements and they are always ordered in the way that depends on UI is not great overall.